api/v1alpha1/selfnoderemediationconfig_webhook.go

```diff
@@ -172,3 +175,26 @@ func validateToleration(toleration v1.Toleration) error {
 	}
 	return nil
 }
+
+func (r *SelfNodeRemediationConfig) validateMinRebootTime() error {
+	if r.Status.MinSafeTimeToAssumeNodeRebootedSeconds == 0 {
```

Comment:
It's depending on the calculated value of `MinSafeTimeToAssumeNodeRebootedSeconds` .
Initially it was set on an annotation but it didn't make sense to keep this annotation once we populate the Status.

Does it bother you because you consider Status fields user informational only ?

>see comment below

Not sure which one do you mean :sweat_smile:

Reply:
I was referring to this comment ("below" as "in a comment for a later line of code 😉"  ) : https://github.com/medik8s/self-node-remediation/pull/197#discussion_r1601215301

It's weird that validation fails because a status field (or annotation, doesn't matter) isn't set yet, isn't it? 🤔
I think this might be a usecase for the new warning, which can be returned instead of an error. That needs some dependency updates though, IIUC (see new method signature e.g. here: https://github.com/medik8s/node-healthcheck-operator/blob/9d59a0387a11c4d38ee45f8fb055a37727e02b74/api/v1alpha1/nodehealthcheck_webhook.go#L68)

Reply:
>It's weird that validation fails because a status field (or annotation, doesn't matter) isn't set yet, isn't it? 🤔

I don't think so, IMO we should consider two main factors:
- Why it wasn't set: In our case this field is set when the agent DS is started so in case it's not set this is an indication that something went very wrong
- The implication of this field being empty:  in our case it means that we can't safely determine a "Safe" value for `Spec.SafeTimeToAssumeNodeRebootedSeconds` , IIUC using an unsafe value could have very bad implications.


>I think this might be a usecase for the new warning, which can be returned instead of an error. 

Hmm good to know ! I didn't even consider the warning  when I've added that piece of code - I just needed a customized validator that can has access to the client.
Do we actually use the warning anywhere in the code ? any idea what does it do as far as user experience ?